media/common/src/main/java/io/helidon/media/common/MediaSupport.java

```diff
@@ -61,27 +90,62 @@ public static MediaSupport createWithDefaults() {
         return builder().registerDefaults().build();
     }
 
+    /**
+     * Create a new instance with the default readers and writers registered on
+     * the contexts.
+     * @param config a {@link Config} that will be passed to {@link
+     * Builder#registerDefaults(Config)}
+     * @return MediaSupport
+     */
+    public static MediaSupport createWithDefaults(Config config) {
+        return builder(config).registerDefaults().build();
+    }
+
     /**
      * Create a new {@link Builder} instance.
-     * @return Builder
+     * @return a new {@link Builder}
      */
     public static Builder builder() {
         return new Builder();
     }
 
+    /**
+     * Create a new {@link Builder} instance.
+     * @param config a {@link Config}
+     * @return a new {@link Builder}
+     */
+    public static Builder builder(Config config) {
+        return builder().config(config);
+    }
+
     /**
      * MediaSupport builder.
      */
     public static final class Builder implements io.helidon.common.Builder<MediaSupport> {
 
         private final MessageBodyReaderContext readerContext;
         private final MessageBodyWriterContext writerContext;
+        private boolean serverErrorsIncludeStackTraces;
 
         Builder() {
             readerContext = MessageBodyReaderContext.create();
             writerContext = MessageBodyWriterContext.create();
         }
 
+        /**
+         * Configures this {@link Builder} from the supplied {@link Config}.
+         * @param config a {@link Config}
+         * @return this {@link Builder}
+         */
+        public Builder config(Config config) {
+            if (config == null) {
+                serverErrorsIncludeStackTraces = false;
+            } else {
+                serverErrorsIncludeStackTraces = config.get("serverErrorsIncludeStackTraces").as(Boolean.TYPE).orElse(false);
```

Comment:
Please use
`config("server-errors-include-stack-traces").asBoolean().ifPresent(this::serverErrorsIncludeStackTraces)`

The problem with your approach:
- if the `serverErrorsIncludeStackTraces` is configured by "hand" and then you update it with config, you override the value even though config does not include that configuration key
- the configuration keys should be dash separated, not camel case
- there are shortcuts for primitives and string on config (e.g. use `asBoolean()` rather than `as(Boolean.class)` or `as(Boolean.TYPE)`)
- I do not see a setter for the `serverErrorsIncludeStackTraces` field in builder, add `public Builder serverErrorsIncludeStackTraces(boolean)` method - everything configurable through config should be configurable through a builder.